linux/pipe/ASoC.py

```diff
@@ -41,21 +42,23 @@ def checkAuth(self):
             "Accept": "application/json",
             "Authorization": "Bearer "+self.token
         }
-        resp = requests.get(f"{self.base_url}/api/V2/Account/TenantInfo", headers=headers)
+        resp = requests.get(f"{self.base_url}/api/v4/Account/TenantInfo", headers=headers)
         return resp.status_code == 200
     
-    def generateIRX(self, scanName, appscanBin, stdoutFilePath = "", configFile=None, secret_scanning=False, printio=True):
-        enableSecrets = ""
+    def generateIRX(self, scanName, scan_flag, appscanBin, stdoutFilePath = "", configFile=None, secret_scanning=False, printio=True):
+        #Build scan arguments
+        args = [appscanBin, "prepare", "-n", scanName]
+        if configFile:
+            args.extend(["-c", configFile])
         if secret_scanning:
-            enableSecrets = "--enableSecrets"
+            args.append("--enableSecrets")
```

Comment:
Can we set this based on the value provided? E.g.:
- If SECRET_SCANNING is not specified, we don't add the option.
- If SECRET_SCANNING is set to True, we add the switch --enableSecrets.
- If SECRET_SCANNING is set to False, we add the switch --noSecrets.

Reply:
made this change